operator/v1/types_ingress.go

```diff
@@ -556,6 +572,52 @@ const (
 	AWSNetworkLoadBalancer AWSLoadBalancerType = "NLB"
 )
 
+// AWSSubnets contains a list of references to AWS subnets by
+// ID or name.
+// +kubebuilder:validation:XValidation:rule=`has(self.ids) && has(self.names) ? size(self.ids + self.names) <= 10 : true`,message="the total number of subnets cannot exceed 10"
```

Comment:
This doesn't work if there are only ids or only names.  It is probably caught later in the other CEL, but it seems confusing that there isn't just one rule that works for `id count > 10` or `name count > 10` or `id + name count > 10`

Reply:
Since we already talked about this, I'll let it pass.

Reply:
> This doesn't work if there are only ids or only names

Correct. Just the combination.

> It is probably caught later in the other CEL, but it seems confusing that there isn't just one rule that works for id count > 10 or name count > 10 or id + name count > 10

Yes, it is caught by the `// +kubebuilder:validation:MaxItems=10` CRD validation, since it's [mandatory](https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#append-only-list-of-containers).

I think you are suggesting this CEL to catch everything:
```
(!(has(self.ids)) || size(self.ids) < 10) &&
(!(has(self.names)) || size(self.names) < 10) &&
(!(has(self.ids) && has(self.names)) || size(self.ids + self.names) < 10)
```

I don't know if adds value to make the CEL more complex here, because we cannot remove `MaxItems` due to [cost-budgeting](https://kubernetes.io/docs/reference/using-api/cel/#runtime-cost-budget). Since we can't remove `MaxItems`, adding these additional CEL checks is redundant. I'm not a API server expert by any means, but I'd imagine avoiding redundant validation is probably preferred here.